### PR TITLE
Align s390x special boot handling to other architectures

### PR DESCRIPTION
**Pull request by OleksandrOrlov**
The PR is made to align s390x kvm special boot handling to other architectures to have less specialization where not needed. It adjusts 'reconnect_s390' subroutine to always wait for grub menu and send 'ret'.

- Related ticket: https://progress.opensuse.org/issues/26836
- Verification run: http://10.160.65.138/tests/169
- Needles: https://gitlab.suse.de/openqa/os-autoinst-needles-sles/merge_requests/882

### CODE REVIEW

File: tests/installation/disable_grub_timeout.pm [hunk 1]
```diff
@@ -21,15 +21,6 @@ use version_utils qw(is_sle is_leap);
 sub run {
     my ($self) = shift;
 
-    if (check_var('ARCH', 's390x')) {
```
Comment (rwx788) [inline tests/installation/disable_grub_timeout.pm]: We don't cover zVM here, let's either enable it for svirt backend only, or address zVM too. 

@Soulofdestiny WDYT?

Comment (OleksandrOrlov) [reply tests/installation/disable_grub_timeout.pm]: zVM was already covered. The task was to implement it for kvm also.

File: lib/utils.pm [hunk 2]
```diff
@@ -1196,13 +1196,9 @@ sub reconnect_s390 {
         select_console('iucvconn');
     }
     else {
-        my $r = wait_serial($login_ready, 300);
-        if ($r && $r =~ qr/Welcome to SUSE Linux Enterprise 15/) {
-            record_soft_failure('bsc#1040606');
```
Comment (okurz) [inline lib/utils.pm]: The bug is RESOLVED FIXED. Could you please make it VERIFIED FIXED before removing this soft fail reference?

File: lib/utils.pm [hunk 3]
```diff
@@ -1196,13 +1196,9 @@ sub reconnect_s390 {
...
-        }
-        elsif ($r && is_sle) {
-            $r =~ qr/Welcome to SUSE Linux Enterprise Server/ || die "Correct welcome string not found";
```
Comment (okurz) [inline lib/utils.pm]: @Soulofdestiny When removing this line but instead just looking for the login prompt further down aren't we loosing some "boot up failure investigation"? I guess we should not just make this bis step between the grub menu and the login prompt but instead look for the intermediate steps as well. WDYT?

**Comment by okurz** [comment]
@OleksandrOrlov could you please attend my comments?

File: lib/utils.pm (hunk 2 above)
Comment (OleksandrOrlov) [reply lib/utils.pm]: As we've already discussed with you some time ago, the ticket number, mentioned on 1201 line is wrong and not related to the workaround with login message on s390x. 

Also, I don't know why that workaround itself was added as it was never executed after adding. Looks like an attempt to make a workaround for bsc#1083506.

**Comment by rwx788** [comment]
As long as we are done with SLE15, we can merge it and see how it works.

File: lib/utils.pm (hunk 3 above)
Comment (Soulofdestiny) [reply lib/utils.pm]: hm... yeah that might happen

**Comment by Soulofdestiny** [comment]
I'd suggest to test this properly before we merge it
there are many different cases to take in account

or just conduct the experiment to merge it and see what breaks =)

**Comment by okurz** [comment]
@OleksandrOrlov it was mentioned by @Soulofdestiny "to test this properly". Have you conducted any further tests? Please cover some more scenarios in the PR description so that we can go ahead

**Comment by OleksandrOrlov** [comment]
I've executed a test here: http://10.160.65.138/tests/169

The last module (svirt_upload_assets) seems to be failed due to network issues.

Also, this PR requires new Needles. The PR: https://gitlab.suse.de/openqa/os-autoinst-needles-sles/merge_requests/882

**Pull request merged**